crates/sui/src/client_commands.rs

```diff
@@ -2215,6 +2215,53 @@ pub(crate) async fn compile_package(
         run_bytecode_verifier,
         print_diags_to_stderr,
     )?;
+    let protocol_config = read_api.get_protocol_config(None).await?;
+
+    // Check that the package's Move version is compatible with the chain's
+    if let Some(Some(SuiProtocolConfigValue::U32(min_version))) = protocol_config
+        .attributes
+        .get("min_move_binary_format_version")
+    {
+        for module in compiled_package.get_modules_and_deps() {
+            if module.version() < *min_version {
+                return Err(SuiError::ModulePublishFailure {
+                    error: format!(
+                        "Module {} has a version {} that is \
+                         lower than the minimum version {min_version} supported by the chain.",
+                        module.self_id(),
+                        module.version(),
+                    ),
+                }
+                .into());
```

Comment:
Not familiar (yet) with `SuiError`. Do we need that or can we just use `bail!`?
```suggestion
                return bail!(
                        "Module {} has a version {} that is \
                         lower than the minimum version {min_version} supported by the chain.",
                        module.self_id(),
                        module.version(),
                    );
              
```

Reply:
Also, what should the user do in this case? Is it possible to give some suggestion on how to go about this? (I actually don't know what to do if I'd see this error).

Reply:
We can use either `bail!` or `SuiError::ModulePublishFailure` although I have a preference for the latter since it's already used below in this function for similar types of errors.

In terms of what the user can do in this case it's very dependent on the error:
* If the version is too low -- this should never happen with our toolchains (ever) so not sure if there's much for us to say there in terms of help.
* If the version is too high -- the advice will then depend a lot on what version it is they're trying to use. You can see an example of this for version 7 below here.